### PR TITLE
feat(core/spinner): Consolidate spinners

### PR DESCRIPTION
**Pull request by caseyhebebrand**
The first in a series of PRs to consolidate Spinners. This PR: 
- Updates `Spinner` with a `vertical` and `circular` mode. The vertical mode is the existing loading bars. The circular mode uses a loading circle.
- Updates the `Spinners`'s angular converter with new props
- Replaces `ButtonBusyIndicator` and `LegacySpinner` with a temporary component for backwards compatibility
- Replaces angular `button-busy-indicator` in core module

Upcoming, once this merges and core is bumped:
- In other packages, replace both Angular and React occurrences of `ButtonBusyIndicator` and `LegacySpinner` with circular `Spinner`
- Remove legacy components from `NgReact` once they are no longer used
- Remove controllers and html for the legacy components

### CODE REVIEW

File: app/scripts/modules/core/src/reactShims/ngReact.ts [hunk 1]
```diff
@@ -41,12 +39,12 @@ export class NgReactInjector extends ReactInject {
 
   // Reactified components
   public AccountRegionClusterSelector: React.ComponentClass<IAccountRegionClusterSelectorProps> = angular2react('accountRegionClusterSelectorWrapper', accountRegionClusterSelectorWrapperComponent, this.$injectorProxy) as any;
-  public ButtonBusyIndicator: React.ComponentClass<IButtonBusyIndicatorProps>                   = angular2react('buttonBusyIndicator', buttonBusyIndicatorComponent, this.$injectorProxy) as any;
+  public ButtonBusyIndicator: React.FunctionComponent<{}>                               = ButtonBusyIndicatorComponent;
```
Comment (christopherthielen) [inline app/scripts/modules/core/src/reactShims/ngReact.ts:42]:
```suggestion
  public ButtonBusyIndicator: React.FunctionComponent<{}>                                       = ButtonBusyIndicatorComponent;
```

**Review by christopherthielen — APPROVED** [review]
❤️

File: app/scripts/modules/core/src/widgets/spinners/Spinner.tsx [hunk 2]
```diff
@@ -1,15 +1,35 @@
...
+  mode?: 'circular' | 'horizontal';
   fullWidth?: boolean;
+  size?: 'nano' | 'small' | 'medium' | 'large';
 }
 
-export class Spinner extends React.Component<ISpinnerProps> {
-  public getBarRows(): React.ReactNode[] {
-    const { size } = this.props;
+export const Spinner = ({
+  color = '#ffffff',
```
Comment (ranihorev) [inline app/scripts/modules/core/src/widgets/spinners/Spinner.tsx:15]: suggest using `var(--color-white)` instead

File: app/scripts/modules/core/src/widgets/spinners/Spinner.tsx [hunk 3]
```diff
@@ -1,15 +1,35 @@
...
+  mode = 'horizontal',
+  size = 'small',
+}: ISpinnerProps) => {
+  if (mode === 'circular') {
+    const sizeToHeight = {
+      nano: 12,
+      small: 16,
+      medium: 24,
+      large: 32,
+    };
```
Comment (ranihorev) [inline app/scripts/modules/core/src/widgets/spinners/Spinner.tsx:27]: I usually prefer to pull maps like this outside of the component and type them

File: app/scripts/modules/core/src/widgets/spinners/Spinner.tsx [hunk 4]
```diff
@@ -26,27 +46,25 @@ export class Spinner extends React.Component<ISpinnerProps> {
...
-
-    return (
-      <div className={classnames('load', size || 'small', { 'full-width': fullWidth })}>
-        {messageNode}
-        {bars}
-      </div>
-    );
-  }
-}
+  const messageClassNames = `message color-text-accent ${size === 'medium' ? 'heading-4' : 'heading-2'}`;
```
Comment (ranihorev) [inline app/scripts/modules/core/src/widgets/spinners/Spinner.tsx:51]: I know that it was here before but it's kind of odd that only `medium` is using a smaller heading

File: app/scripts/modules/core/src/widgets/spinners/Spinner.tsx (hunk 2 above)
Comment (ranihorev) [reply app/scripts/modules/core/src/widgets/spinners/Spinner.tsx:15]: btw, should we use the color for the bars as well?

**Pull request merged**